shell/common/shell.cc

```diff
@@ -816,6 +816,12 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
+  if (metrics.device_pixel_ratio <= 0) {
```

Comment:
I think this should also check the width and height for positivity, or a 2nd check could handle that.  Otherwise, there seems to be nothing in the engine guarding against negative or 0 dimensions

Reply:
IIRC a zero sized viewport just disengages the animator and is a valid case. That is also the initial viewport size.

Reply:
Done, and just updated the same test case.

Reply:
It is the initial size, but I'm not seeing where it disengages the animator.

What would it mean to actually set the viewport to zero width or height?

Reply:
Hmm, you're right. I thought `Shell::OnPlatformViewSetViewportMetrics` used to pause the animator when the size was zero. That doesn't not seem to be the case.

Reply:
The skips rendering the generated LayerTree here:  https://github.com/flutter/engine/blob/master/shell/common/engine.cc#L483

Reply:
Also note that LayerTree has another DCHECK for non-zero DPR:  https://github.com/flutter/engine/blob/master/flow/layers/layer_tree.cc#L20

Reply:
That one's a CHECK. That'll just crash the whole process in release mode. It's probably .. ok to crash there rather than later.